compiler/src/iree/compiler/GlobalOptimization/SetEncoding.cpp

```diff
@@ -284,6 +268,26 @@ struct setContractionOpEncoding
       return rewriter.notifyMatchFailure(
           linalgOp, "the op has preset compilation strategy, skip SetEncoding");
     }
+    if (!linalg::isaContractionOpInterface(linalgOp)) {
+      return rewriter.notifyMatchFailure(linalgOp,
+                                         "the op is not a contraction op");
+    }
+    auto maps = linalgOp.getIndexingMaps();
+    auto cDims = linalg::inferContractionDims(linalgOp);
+    if (failed(cDims) || cDims->batch.size() > 1 || cDims->m.size() > 1 ||
+        cDims->n.size() > 1 || cDims->k.size() > 1) {
+      return rewriter.notifyMatchFailure(
+          linalgOp, "Expected {|Batch|, |M|, |N|, |K|} <= 1");
+    }
+    if ((cDims->n.empty() && cDims->m.empty()) || cDims->k.empty()) {
+      return rewriter.notifyMatchFailure(
+          linalgOp, "Expected M or N dims and K dim to not be empty");
+    }
+    if (!hasMatmulLikeBody(linalgOp)) {
+      return rewriter.notifyMatchFailure(
+          linalgOp, "Expected op to have a matmul body, i.e. yield(add(out, "
+                    "mul(cast(in0), cast(in1))))");
```

Comment:
I thought this is checked by `isaContractionOpInterface`?

Reply:
I added this because I noticed that the `isaContractionOpInterface` has been generalized a bit to different types of ops since I last looked at it:
https://github.com/llvm/llvm-project/blob/55cb52bbc539777aeb176f22fef0b4bfc70883fc/mlir/lib/Dialect/Linalg/IR/LinalgInterfaces.cpp#L323-L330
It now has complex add/mul, as well as and/or, and I don't think we can support these cases right now. I will add a TODO comment with an issue about this though. I'm not sure if we see any such cases, but it could be good to further extend data tiling to start being able to create `linalg.generic` ops instead of only `mmt4d` in the future.

Reply:
I see, thanks for the details! Please add a TODO (or at least a comment).